### DISCUSSION TITLE
Stuck with evaluation for an newly created exercise

### DISCUSSION THREAD

**Discussion by marcothijs** [dodona-edu/feedback-and-support#17] [Questions - Vragen] [answered]
I need some help in evaluating the following exercise: https://dodona.ugent.be/nl/activities/1442631305/
It only has a Dutch description but look at "Voorbeeld" to get the idea. 
Running the program, is shows immediately the 2D list (dots show as a matrix), before any input. Then, after each input it show the updated matrix till the user enters a "0" (zero) to end the program and show a last text output. 
My attempts didn't give me the correct evaluation so I had to take it out. 
Thanks for your help!
Marco.

**Comment by chvp** [comment, marked as answer]
What problems did you encounter? When using the input-output python judge, I would recommend using both `ends with` for `input block size` and `output block size` for this exercise. Another thing that is also important, is that output that is printed by `input()` is ignored by the judge by default, so `Speler 1: 'o' -> kolom (1 - 7) - (0 om te stoppen): ` should not be part of the expected output. Hope this helps.

**Reply by marcothijs** [reply]
The difference with my other exercises is that the code starts with printing a matrix (with dots, see example in description) without any input given in the first place. The program then will get an input (column number) followed by an output (again a matrix) till the game ends with a last output (a string), this all in 1 program. I've added the evaluation again so you can see what I've tried so far. 

Thanks for your help.

**Reply by chvp** [reply]
If I remove the empty lines in the evaluation files that are in your repository right now, the sample solution gets evaluated correctly. `ends with` makes it that each empty line is seen as a new testcase, so if you only want one testcase with all those lines as expected input and output, there can be no empty lines it the test files.

**Reply by marcothijs** [reply]
I've removed all the empty lines in the evalutation files but it still doesn't get evaluated correctly. See print screen. If you have the right to change the evaluation files, please go ahead. 
<img width="717" alt="Screenshot 2023-02-27 at 17 35 26" src="https://user-images.githubusercontent.com/60760890/221624231-5fd3128e-a0b0-4ff0-95d8-53d80e2ffb24.png">

**Reply by chvp** [reply]
The files are the same as I have them. Are you submitting the sample solution? (The file `solution/solution.nl.py`)

If I submit that with the current files it gets evaluated as correct.

**Reply by marcothijs** [reply]
I guess I didn't ... Took the last submission ... So you're right, it works now. Thanks for your help (and patience ;-))
